builtin/plugins/dposv3/dpos_test.go

```diff
@@ -1299,6 +1299,165 @@ func TestClaimRewardsFromMultipleValidators(t *testing.T) {
+	tier := uint64(1)
+	for _, addr := range addrs {
+		err = coinContract.Approve(contractpb.WrapPluginContext(coinCtx.WithSender(delegatorAddress1)), &coin.ApproveRequest{
+			Spender: dpos.Address.MarshalPB(),
+			Amount:  &types.BigUInt{Value: delegationAmount},
+		})
+		require.Nil(t, err)
+
+		err = dpos.Delegate(pctx.WithSender(delegatorAddress1), &addr, delegationAmount.Int, &tier, nil)
+		require.Nil(t, err)
+	}
+
+	// Do a bunch of elections that correspond to 1/12th of a year
+	for i := int64(0); i < yearSeconds/12; i = i + cycleLengthSeconds {
+		require.NoError(t, elect(pctx, dpos.Address))
+		pctx.SetTime(pctx.Now().Add(time.Duration(cycleLengthSeconds) * time.Second))
+	}
+
+	candidates, err := dpos.ListCandidates(pctx)
+	require.Nil(t, err)
+	assert.Equal(t, 3, len(candidates))
+
+	err = dpos.UnregisterCandidate(pctx.WithSender(addr1))
+	require.Nil(t, err)
+
+	candidates, err = dpos.ListCandidates(pctx)
+	require.Nil(t, err)
+	assert.Equal(t, 3, len(candidates))
+
+	validators, err = dpos.ListValidators(pctx)
+	require.Nil(t, err)
+	assert.Equal(t, 3, len(validators))
+
+	require.NoError(t, elect(pctx, dpos.Address))
+	pctx.SetTime(pctx.Now().Add(time.Duration(cycleLengthSeconds) * time.Second))
+
+	rewardFromAddr1Before, err := dpos.CheckRewardDelegation(pctx.WithSender(delegatorAddress1), &addr1)
+	require.NoError(t, err)
+	rewardFromAddr2, err := dpos.CheckRewardDelegation(pctx.WithSender(delegatorAddress1), &addr2)
+	require.NoError(t, err)
+	rewardFromAddr3, err := dpos.CheckRewardDelegation(pctx.WithSender(delegatorAddress1), &addr3)
+	require.NoError(t, err)
+	require.True(t, rewardFromAddr2.Amount.Value.Cmp(&rewardFromAddr3.Amount.Value) == 0)
+	require.True(t, rewardFromAddr2.Amount.Value.Cmp(&rewardFromAddr1Before.Amount.Value) == 0)
+
+	require.NoError(t, elect(pctx, dpos.Address))
+	pctx.SetTime(pctx.Now().Add(time.Duration(cycleLengthSeconds) * time.Second))
+
+	candidates, err = dpos.ListCandidates(pctx)
+	require.Nil(t, err)
+	assert.Equal(t, 2, len(candidates))
+
+	validators, err = dpos.ListValidators(pctx)
+	require.Nil(t, err)
+	assert.Equal(t, 2, len(validators))
+
+	rewardFromAddr1, err := dpos.CheckRewardDelegation(pctx.WithSender(delegatorAddress1), &addr1)
+	require.NoError(t, err)
+	require.True(t, rewardFromAddr1.Amount.Value.Cmp(&rewardFromAddr1Before.Amount.Value) == 0)
+
+	rewardFromAddr2, err = dpos.CheckRewardDelegation(pctx.WithSender(delegatorAddress1), &addr2)
+	require.NoError(t, err)
+	rewardFromAddr3, err = dpos.CheckRewardDelegation(pctx.WithSender(delegatorAddress1), &addr3)
+	require.NoError(t, err)
+	require.True(t, rewardFromAddr2.Amount.Value.Cmp(&rewardFromAddr3.Amount.Value) == 0)
+	require.True(t, rewardFromAddr2.Amount.Value.Cmp(&rewardFromAddr1.Amount.Value) > 0)
+
+	// User claims the rewards they expected with 1 call.
+	// They are also able to get the amount that was claimed in the same call
+	amt, err := dpos.ClaimDelegatorRewards(pctx.WithSender(delegatorAddress1))
```

Comment:
I don't see `ClaimRewardsFromAllValidators` being called anywhere in this test.

Reply:
`ClaimDelegatorRewards` is the test helper that calls `ClaimRewardsFromAllValidators`.